softlayer/resource_softlayer_lb_local.go

```diff
@@ -91,14 +103,28 @@ func resourceSoftLayerLbLocalCreate(d *schema.ResourceData, meta interface{}) er
 		capacity = c
 	}
 
-	var keyname string
-	if haEnabled {
-		keyname = "DEDICATED_LOAD_BALANCER_WITH_HIGH_AVAILABILITY_AND_SSL"
+	var keyFormatter string
+	if dedicated {
+		// Dedicated local LB always comes with SSL support
+		d.Set("ssl_enabled", true)
+		categoryCode = product.DedicatedLoadBalancerCategoryCode
+		if haEnabled {
+			keyFormatter = "DEDICATED_LOAD_BALANCER_WITH_HIGH_AVAILABILITY_AND_SSL_%d_CONNECTIONS"
+		} else {
+			keyFormatter = "LOAD_BALANCER_DEDICATED_WITH_SSL_OFFLOAD_%d_CONNECTIONS"
+		}
 	} else {
-		keyname = "LOAD_BALANCER_DEDICATED_WITH_SSL_OFFLOAD"
+		categoryCode = product.ProxyLoadBalancerCategoryCode
+		if _, ok := d.GetOk("security_certificate_id"); ok {
+			d.Set("ssl_enabled", true)
+			keyFormatter = "LOAD_BALANCER_%d_VIP_CONNECTIONS_WITH_SSL_OFFLOAD"
+		} else {
+			d.Set("ssl_enabled", false)
+			keyFormatter = "LOAD_BALANCER_%d_VIP_CONNECTIONS"
+		}
```

Comment:
`ha_enabled` is a required field and shared load balancers don't support HA. It is necessary to return an error if `dedicated = false` and `ha_enabled = true`  in a create function.